vyper/exceptions.py

```diff
@@ -78,3 +78,15 @@ class FunctionDeclarationException(ParserException):
 
 class EventDeclarationException(ParserException):
     pass
+
+
+class ContractException(Exception):
```

Comment:
RE: handling multiple files with `vyper.compile` - I don't think this is the clearest way to solve this. Can handle exceptions in the script and log them there. "issue while compiling X:"

Reply:
I think actually making the `compile` interface return only a data structure on compilation faults might be even cleaner for integration of the compiler module?

Reply:
See below commit, I think this solves both scenarios ;)

Reply:
Yeah, I like that much better